modules/library/src/main/scala/zio/elasticsearch/HttpElasticExecutor.scala

```diff
@@ -37,23 +40,35 @@ private[elasticsearch] final class HttpElasticExecutor private (config: ElasticC
 
   import HttpElasticExecutor._
 
-  def execute[A](request: ElasticRequest[A, _]): Task[A] =
+  def execute[A](request: ElasticRequest[A]): Task[A] =
     request match {
-      case r: BulkRequest           => executeBulk(r)
-      case r: CreateRequest         => executeCreate(r)
-      case r: CreateWithIdRequest   => executeCreateWithId(r)
-      case r: CreateIndexRequest    => executeCreateIndex(r)
-      case r: CreateOrUpdateRequest => executeCreateOrUpdate(r)
-      case r: DeleteByIdRequest     => executeDeleteById(r)
-      case r: DeleteByQueryRequest  => executeDeleteByQuery(r)
-      case r: DeleteIndexRequest    => executeDeleteIndex(r)
-      case r: ExistsRequest         => executeExists(r)
-      case r: GetByIdRequest        => executeGetById(r)
-      case r: GetByQueryRequest     => executeGetByQuery(r)
-      case map @ Map(_, _)          => execute(map.request).flatMap(a => ZIO.fromEither(map.mapper(a)))
+      case r: Bulk           => executeBulk(r)
+      case r: Create         => executeCreate(r)
+      case r: CreateWithId   => executeCreateWithId(r)
+      case r: CreateIndex    => executeCreateIndex(r)
+      case r: CreateOrUpdate => executeCreateOrUpdate(r)
+      case r: DeleteById     => executeDeleteById(r)
+      case r: DeleteByQuery  => executeDeleteByQuery(r)
+      case r: DeleteIndex    => executeDeleteIndex(r)
+      case r: Exists         => executeExists(r)
+      case r: GetById        => executeGetById(r)
+      case r: GetByQuery     => executeGetByQuery(r)
```

Comment:
@markaya please replace `GetByQuery` with `Search` everywhere to align with the name `SearchResult` and the naming coming in the PR for aggregations.